### DISCUSSION TITLE
Python formatted print function is not supported in Mojo

### DISCUSSION THREAD

**Discussion by bennylam** [modular/modular#539] [Q&A] [answered]
I have tested some python combability codes in Mojo Notebook and found the formatted print function seems not supported in Mojo.
Is any combability list of Python syntax and functions available for Mojo?

My test code and error output is shown below:
====================
from python import Python
def fact(x):
    n = 1
    for i in range(x):
        n = n*(i+1)
    print(f'you entered {x} and the Factorial is:',n)  # Python formatted print function is not supported in Mojo??

Error Output:
error: Expression [38]:10:12: expected ')' in call argument list
    print(f'you entered {x} and the Factorial is:',n)  # Python formatted print function is not supported in Mojo??
           ^

expression failed to parse (no further compiler diagnostics)

**Comment by drunkwcodes** [comment, marked as answer]
@bennylam This is disussed in https://github.com/modularml/mojo/issues/398
Welcome to join that issue.

P.S Please use code block syntax "```" when you want to paste code to reserve the correct format.

**Comment by yury46** [comment]
I created a useful **konus.py** program for calculating the parameters of the trunk and log (volume, weight, price,...), I want to publish an executable file. But konus.mojo cannot be compiled because MOJO does not fully support the syntax of the print () function. I have 2 questions:

1. When will MOJO fully support the syntax of the print () function so that you do not have to enter var, etc.?

To allow print ('Diameter D ='% .1f '% D,' (dm) ')

2. When will MOJO be available on Windows 10 and 11 without the need for WSL?
